modules/workergroup/locals.tf

```diff
@@ -0,0 +1,95 @@
+# Copyright 2022, Oracle Corporation and/or affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+locals {
+  tenancy_id = coalesce(var.tenancy_id, var.tenancy_ocid)
+}
+
+locals {
+  cluster = lookup({ # Look for existing cluster by OCID, used in subsequent locals blocks
+    for c in lookup(data.oci_containerengine_clusters.wg_clusters, "clusters", []) : c.id => c if c.id == var.cluster_id
+  }, var.cluster_id, {})
+  compartment_id   = coalesce(var.worker_compartment_id, var.compartment_id, var.compartment_ocid, local.tenancy_id)
+  region           = coalesce(var.region, "us-ashburn-1")
```

Comment:
The default should be the home_region. Some users may choose not to subscribe to any other regions other than their tenancy's home region.